packages/pangea-sdk/src/test/java/cloud/pangeacyber/pangea/vault/ITVaultTest.java

```diff
@@ -695,20 +707,34 @@ public void testExport() throws PangeaException, PangeaAPIException {
 		final var key = generated.getResult().getId();
 		assertNotNull(key);
 
+		// Generate a RSA key pair.
+		final var keyPair = CryptoUtils.generateRsaKeyPair(4096);
+		final var publicKey = CryptoUtils.asymmetricPemExport(keyPair.getPublic());
+
 		// Export it.
-		var request = new ExportRequest.Builder(key).build();
+		var request = new ExportRequest.Builder(key)
+			.encryptionAlgorithm(ExportEncryptionAlgorithm.RSA4096_OAEP_SHA512)
+			.encryptionKey(publicKey)
+			.build();
 		var actual = client.export(request);
 		assertNotNull(actual);
 		assertEquals(key, actual.getResult().getId());
-		assertNotNull(actual.getResult().getPublicKey());
-		assertNotNull(actual.getResult().getPrivateKey());
+		assertTrue(actual.getResult().isEncrypted());
+		final var decodedPublicKey = (new Base64(true)).decode(actual.getResult().getPublicKey());
+		final var cipher = Cipher.getInstance("RSA/ECB/OAEPWithSHA-512AndMGF1Padding", new BouncyCastleProvider());
+		assertEquals(
+			generated.getResult().getEncodedPublicKey(),
+			new String(CryptoUtils.asymmetricDecrypt(cipher, keyPair.getPrivate(), decodedPublicKey))
+		);
+		final var decodedPrivateKey = (new Base64(true)).decode(actual.getResult().getPrivateKey());
+		final var decryptedPrivateKey = CryptoUtils.asymmetricDecrypt(cipher, keyPair.getPrivate(), decodedPrivateKey);
```

Comment:
IMO it would be better to generate `cipher` inside `asymmetricDecrypt`. I know we would be doing a less flexible function, but it should be as much easy to use as possible to end user. I wouldn't want them to have to go to the examples each time they want to decrypt a key just to copy/past this string "RSA/ECB/OAEPWithSHA-512AndMGF1Padding"

Reply:
Kinda falls apart once Vault supports more than one export encryption algorithm, although we could determine the cipher if we have the user pass the `ExportEncryptionAlgorithm` enum value they used earlier. What do you think of that?

Reply:
Yes, I like that!